pkg/pipeline/remote/gitlab/gitlab.go

```diff
@@ -474,25 +469,24 @@ func doRequestToGitlab(method string, url string, gitlabAccessToken string, opt
 	return resp, nil
 }
 
-func paginateGitlab(gitlabAccessToken string, url string) ([]*http.Response, error) {
-	var responses []*http.Response
-
-	response, err := getFromGitlab(gitlabAccessToken, url)
-	if err != nil {
-		return responses, err
-	}
-	responses = append(responses, response)
-	nextURL := nextGitlabPage(response)
+func paginateGitlab(gitlabAccessToken string, url string) ([][]byte, error) {
+	var responseBodies [][]byte
+	var nextURL = url
 	for nextURL != "" {
-		response, err = getFromGitlab(gitlabAccessToken, nextURL)
+		response, err := getFromGitlab(gitlabAccessToken, nextURL)
+		if err != nil {
+			return nil, err
+		}
+		body, err := ioutil.ReadAll(response.Body)
 		if err != nil {
-			return responses, err
+			return nil, err
 		}
-		responses = append(responses, response)
+		response.Body.Close()
```

Comment:
should we add `defer response.Body.Close()` before `ReadAll` in case that returns an err?

Reply:
It is in a for loop, therefore `Close()` is called directly instead of using defer. Updated to call `Close()` on `readAll` errors.